homeassistant/components/jewish_calendar/sensor.py

```diff
@@ -21,146 +22,212 @@
+    """Class describing Jewish Calendar sensor entities."""
+
+    value_fn: Callable[[JewishCalendarDataResults], str | int]
+    attr_fn: Callable[[JewishCalendarDataResults], dict[str, str]] | None = None
+    options_fn: Callable[[bool], list[str]] | None = None
+
+
+@dataclass(frozen=True, kw_only=True)
+class JewishCalendarTimestampSensorDescription(JewishCalendarBaseSensorDescription):
+    """Class describing Jewish Calendar sensor timestamp entities."""
+
+    value_fn: (
+        Callable[[HDateInfo, Callable[[dt.date], Zmanim]], dt.datetime | None] | None
+    ) = None
+
+
+INFO_SENSORS: tuple[JewishCalendarSensorDescription, ...] = (
+    JewishCalendarSensorDescription(
         key="date",
         translation_key="hebrew_date",
         icon="mdi:star-david",
+        value_fn=lambda results: str(results.after_shkia_date.hdate),
+        attr_fn=lambda results: {
+            "hebrew_year": str(results.after_shkia_date.hdate.year),
+            "hebrew_month_name": str(results.after_shkia_date.hdate.month),
+            "hebrew_day": str(results.after_shkia_date.hdate.day),
+        },
     ),
-    SensorEntityDescription(
+    JewishCalendarSensorDescription(
         key="weekly_portion",
         translation_key="weekly_portion",
         icon="mdi:book-open-variant",
         device_class=SensorDeviceClass.ENUM,
+        options_fn=lambda _: [str(p) for p in Parasha],
+        value_fn=lambda results: str(results.after_tzais_date.upcoming_shabbat.parasha),
     ),
-    SensorEntityDescription(
+    JewishCalendarSensorDescription(
         key="holiday",
         translation_key="holiday",
         icon="mdi:calendar-star",
         device_class=SensorDeviceClass.ENUM,
+        options_fn=lambda diaspora: HolidayDatabase(diaspora).get_all_names(),
+        value_fn=lambda results: ", ".join(
+            str(holiday) for holiday in results.after_shkia_date.holidays
+        ),
+        attr_fn=lambda results: {
+            "id": ", ".join(
+                holiday.name for holiday in results.after_shkia_date.holidays
+            ),
+            "type": ", ".join(
+                dict.fromkeys(
+                    _holiday.type.name for _holiday in results.after_shkia_date.holidays
+                )
+            ),
+        },
     ),
-    SensorEntityDescription(
+    JewishCalendarSensorDescription(
         key="omer_count",
         translation_key="omer_count",
         icon="mdi:counter",
         entity_registry_enabled_default=False,
+        value_fn=lambda results: results.after_shkia_date.omer.total_days
+        if results.after_shkia_date.omer
+        else 0,
     ),
-    SensorEntityDescription(
+    JewishCalendarSensorDescription(
         key="daf_yomi",
         translation_key="daf_yomi",
         icon="mdi:book-open-variant",
         entity_registry_enabled_default=False,
+        value_fn=lambda results: str(results.daytime_date.daf_yomi),
```

Comment:
what's the type of this?

Reply:
That's a dataclass called DafYomi. The __str__ method returns the value

Reply:
That sounds strange, What's the stringified value?

Reply:
It represents a tractate of the talmud and a specific page. The stringified value concatenates the two fields (either in Hebrew or transliterated). In Hebrew the numbering is represented by Hebrew letters. If transliterated, it's represented by numbers.